plugins/services/watchdog/service.go

```diff
@@ -0,0 +1,74 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package watchdog
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/containerd/containerd/v2/plugins"
+	"github.com/containerd/plugin"
+	"github.com/containerd/plugin/registry"
+	"github.com/coreos/go-systemd/v22/daemon"
+)
+
+func init() {
+	registry.Register(&plugin.Registration{
+		Type:     plugins.WatchdogPlugin,
+		ID:       "software-watchdog",
+		Requires: []plugin.Type{},
+		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
+			if runtime.GOOS != "linux" {
+				return nil, fmt.Errorf("host does not support watchdog: %w", plugin.ErrSkipPlugin)
+			}
+			watchdogUsec := os.Getenv("WATCHDOG_USEC")
```

Comment:
Is this an existing variable set by another process? If its for configuration use the existing configuration, otherwise document where this env comes from.

Reply:
$WATCHDOG_USEC is set by the system manager for the process which has watchdog support enabled.

Reply:
It'd help to link to https://0pointer.de/blog/projects/watchdog.html and/or https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html to describe where WATCHDOG_USEC comes from.

Reply:
ok, I took another pass on this. I don't think we need to parse watchdog related env here. [`daemon.SdWatchdogEnabled`](https://github.com/coreos/go-systemd/blob/7d375ecc2b092916968b5601f74cca28a8de45dd/daemon/watchdog.go#L35) already does this for us.